subprojects/testing-base/src/main/java/org/gradle/api/internal/tasks/testing/junit/result/JUnitXmlResultWriter.java

```diff
@@ -41,7 +41,7 @@ public JUnitXmlResultWriter(String hostName, TestResultsProvider testResultsProv
      * @param output The destination, unbuffered
      */
     public void write(TestClassResult result, OutputStream output) {
-        String className = result.getClassDisplayName();
+        String className = result.hasDefaultDisplayName() ? result.getClassName() : result.getClassDisplayName();
```

Comment:
I think this logic belongs inside `TestClassResult`. It could expose a method like `getDisplayName()` that does exactly this.

Reply:
I think that would be confusing because there's already `getClassDisplayName()`. How about `getXmlTestSuiteName()`?

Reply:
I was thinking about that as well, but I don't think the `TestClassResult` should know about the xml report. OTOH they both live in the same package so the argument is not that strong...

Reply:
What about something like `getReportingDisplayName`?

Reply:
It's only used in case of the XML report, though. It could be simplified if we decided to always use the display name for the `<testsuite name="...">` attribute.

Reply:
As discussed, I went with `getXmlTestSuiteName()`.